mobile/src/main/java/org/openhab/habdroid/core/connection/ConnectionAvailbilityAwareAcivity.java

```diff
@@ -0,0 +1,74 @@
+package org.openhab.habdroid.core.connection;
+
+import android.app.Activity;
+import android.content.Intent;
+import android.os.Bundle;
+import android.support.annotation.Nullable;
+import android.support.v4.app.TaskStackBuilder;
+import android.support.v7.app.AppCompatActivity;
+
+import org.openhab.habdroid.core.connection.exception.NetworkNotAvailableException;
+import org.openhab.habdroid.core.connection.exception.NetworkNotSupportedException;
+import org.openhab.habdroid.core.connection.exception.NoUrlInformationException;
+import org.openhab.habdroid.ui.NoNetworkActivity;
+import org.openhab.habdroid.ui.OpenHABPreferencesActivity;
+
+import static org.openhab.habdroid.ui.NoNetworkActivity.NO_NETWORK_MESSAGE;
+import static org.openhab.habdroid.ui.OpenHABPreferencesActivity.NO_URL_INFO_EXCEPTION_EXTRA;
+import static org.openhab.habdroid.ui.OpenHABPreferencesActivity.NO_URL_INFO_EXCEPTION_MESSAGE;
+
+public abstract class ConnectionAvailbilityAwareAcivity extends AppCompatActivity {
+    private Thread.UncaughtExceptionHandler originalHandler;
+
+    @Override
+    protected void onCreate(@Nullable Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+
+        setConnectionExceptionHandler();
+    }
+
+    @Override
+    protected void onDestroy() {
+        super.onDestroy();
+
+        if (originalHandler == null) {
+            return;
+        }
+
+        Thread.setDefaultUncaughtExceptionHandler(originalHandler);
+    }
+
+    private void setConnectionExceptionHandler() {
+        final Activity activity = this;
+
+        originalHandler = Thread.getDefaultUncaughtExceptionHandler();
+
+        Thread.setDefaultUncaughtExceptionHandler(new Thread.UncaughtExceptionHandler() {
+            @Override
+            public void uncaughtException(Thread thread, Throwable throwable) {
+                if (throwable instanceof NetworkNotAvailableException) {
+                    Intent networkNotSupportedIntent = new Intent(activity, NoNetworkActivity.class);
+                    networkNotSupportedIntent.putExtra(NO_NETWORK_MESSAGE, throwable.getMessage());
+                    startActivity(networkNotSupportedIntent);
```

Comment:
Why have a separate activity here instead of just replacing the activity content?

Reply:
Hmm, it should be possible to do that, too. I'm not sure, what the best approach would be here, replacing the fragment?

Reply:
Having our own activity has the advantage that we can control how it looks like (e.g. no voice control, and no up/hamburger menu). We could probably disable these elements in the current activity, too, however, that would make the code here not that independant from the subclass anymore. It also would require to redraw the activity (would it?), so starting a new activity is probably not that much overhead?

Reply:
What do you mean by 'redraw the activity'? Showing and hiding some views is *much* cheaper tahn creating a new activity and doesn't have the side effects of creating a new activity (e.g. animations). See the setErrorViewVisibility() method in above link for how I'd solve this.